src/ArmTemplates/Extractor/Models/ExtractorParameters.cs

```diff
@@ -1,56 +1,133 @@
     {
-        public string sourceApimName { get; private set; }
-        public string destinationApimName { get; private set; }
-        public string resourceGroup { get; private set; }
-        public string fileFolder { get; private set; }
-        public string linkedTemplatesBaseUrl { get; private set; }
-        public string linkedTemplatesSasToken { get; private set; }
-        public string linkedTemplatesUrlQueryString { get; private set; }
-        public string policyXMLBaseUrl { get; private set; }
-        public string policyXMLSasToken { get; private set; }
-        public string apiVersionSetName { get; private set; }
-        public bool includeAllRevisions { get; private set; }
-        public serviceUrlProperty[] serviceUrlParameters { get; set; }
-        public bool paramServiceUrl { get; private set; }
-        public bool paramNamedValue { get; private set; }
-        public bool paramApiLoggerId { get; private set; }
-        public bool paramLogResourceId { get; private set; }
-        public bool notIncludeNamedValue { get; private set; }
-        public bool paramNamedValuesKeyVaultSecrets { get; private set; }
-
-        public int operationBatchSize { get; private set; }
-
-        public bool paramBackend { get; set; }
-
-        public ExtractorParameters(ExtractorConfig exc, string dirName)
+        public string SourceApimName { get; private set; }
+
+        public string DestinationApimName { get; private set; }
+
+        public string ResourceGroup { get; private set; }
+
+        /// <summary>
+        /// Naming of all templates from the base folder requested by end-user in extractor-configuration parameters
+        /// </summary>
+        public FileNames FileNames { get; private set; }
+
+        /// <summary>
+        /// Not changeable root directory, where templates will be generated.
+        /// </summary>
+        public string FilesGenerationRootDirectory { get; private set; }
+
+        /// <summary>
+        /// Name of a single API that user wants to extract
+        /// </summary>
+        public string SingleApiName { get; set; }
+
+        /// <summary>
+        /// Names of APIs that user wants to extract
+        /// </summary>
+        public List<string> MultipleApiNames { get; set; }
+
+        /// <summary>
+        /// Create split api templates for every API in the source APIM instance
+        /// </summary>
+        public bool SplitApis { get; private set; }
+
+        /// <summary>
+        /// Are revisions included to templates
+        /// </summary>
+        public bool IncludeAllRevisions { get; private set; }
+
+        public string LinkedTemplatesBaseUrl { get; private set; }
+
+        public string LinkedTemplatesSasToken { get; private set; }
+
+        public string LinkedTemplatesUrlQueryString { get; private set; }
+
+        public string PolicyXMLBaseUrl { get; private set; }
+
+        public string PolicyXMLSasToken { get; private set; }
+
```

Comment:
Is there a reason for having an empty line between every property (without a summary)? It just makes the file unnecessarily long and doesn't make things more readable or anything IMO .

Reply:
For me it is a little bit hard to read the properties characteristics (like type, visibility modifier and etc) when they are located one by one. I think it is just a personal preference.

However for that specific file you've written your comment to (`ExtractorParameters`) I maybe will write summary to every property inside, because it is like a main exposed parameters container for **Extractor**

Resolving this now, if you think it is better to be changed, feel free to say it!